client/keystore/src/lib.rs

```diff
@@ -276,8 +298,65 @@ impl Store {
 }
 
 impl BareCryptoStore for Store {
+	fn supported_keys(
+		&self,
+		id: KeyTypeId,
+		keys: Vec<CryptoTypePublicPair>
+	) -> std::result::Result<HashSet<CryptoTypePublicPair>, TraitError> {
+		let keys = keys.into_iter().collect::<HashSet<_>>();
+		let all_keys = self.keys(id)?;
+
+		Ok(keys.intersection(&all_keys).cloned().collect())
+	}
+
+	fn keys(&self, id: KeyTypeId) -> std::result::Result<HashSet<CryptoTypePublicPair>, TraitError> {
+		let ed25519_existing_keys = self
+			.public_keys_by_type::<ed25519::Public>(id)
+			.map_err(|e| TraitError::from(e))?
+			.into_iter()
+			.map(|k| (ed25519::ED25519_CRYPTO_ID, k.to_raw_vec()));
+
+		let sr25519_existing_keys = self
+			.public_keys_by_type::<sr25519::Public>(id)
+			.map_err(|e| TraitError::from(e))?
+			.into_iter()
+			.map(|k| (sr25519::SR25519_CRYPTO_ID, k.to_raw_vec()));
+
+		Ok(ed25519_existing_keys
+			.chain(sr25519_existing_keys)
+			.collect::<HashSet<_>>())
+	}
+
+	fn sign_with(
+		&self,
+		id: KeyTypeId,
+		key: &CryptoTypePublicPair,
+		msg: &[u8],
+	) -> std::result::Result<Vec<u8>, TraitError> {
+		match key.0 {
+			ed25519::ED25519_CRYPTO_ID => {
+				let pub_key = ed25519::Public::from_slice(key.1.as_slice());
+				let key_pair: ed25519::Pair = self
+					.key_pair_by_type::<ed25519::Pair>(&pub_key, id)
+					.map(Into::into)
+					.map_err(|e| TraitError::from(e))?;
+				return Ok(<[u8; 64]>::from(key_pair.sign(msg)).to_vec());
+			}
+			sr25519::SR25519_CRYPTO_ID => {
+				let pub_key = sr25519::Public::from_slice(key.1.as_slice());
+				let key_pair: sr25519::Pair = self
+					.key_pair_by_type::<sr25519::Pair>(&pub_key, id)
+					.map(Into::into)
+					.map_err(|e| TraitError::from(e))?;
+				return Ok(<[u8; 64]>::from(key_pair.sign(msg)).to_vec());
```

Comment:
```suggestion
				Ok(<[u8; 64]>::from(key_pair.sign(msg)).to_vec())
```

We can drop `return` cause the entire expression is returned anyway.